rust/src/data_catalog/unity/mod.rs

```diff
@@ -75,17 +37,548 @@ pub enum GetTableError {
+                if key.starts_with("UNITY_") || key.starts_with("DATABRICKS_") {
+                    if let Ok(config_key) =
+                        UnityCatalogConfigKey::from_str(&key.to_ascii_lowercase())
+                    {
+                        builder = builder.try_with_option(config_key, value).unwrap();
+                    }
+                }
+            }
+        }
+
+        builder
+    }
+
+    /// Set the URL of a Databricks workspace.
+    pub fn with_workspace_url(mut self, url: impl Into<String>) -> Self {
+        self.workspace_url = Some(url.into());
+        self
+    }
+
+    /// Sets the client id for use in client secret or k8s federated credential flow
+    pub fn with_client_id(mut self, client_id: impl Into<String>) -> Self {
+        self.client_id = Some(client_id.into());
+        self
+    }
+
+    /// Sets the client secret for use in client secret flow
+    pub fn with_client_secret(mut self, client_secret: impl Into<String>) -> Self {
+        self.client_secret = Some(client_secret.into());
+        self
+    }
+
+    /// Sets the authority id for use service principal credential based authentication
+    pub fn with_authority_id(mut self, tenant_id: impl Into<String>) -> Self {
+        self.authority_id = Some(tenant_id.into());
+        self
+    }
+
+    /// Set a static bearer token to be used for authorizing requests
+    pub fn with_bearer_token(mut self, bearer_token: impl Into<String>) -> Self {
+        self.bearer_token = Some(bearer_token.into());
+        self
+    }
+
+    /// Set a personal access token (PAT) to be used for authorizing requests
+    pub fn with_access_token(self, access_token: impl Into<String>) -> Self {
+        self.with_bearer_token(access_token)
+    }
+
+    /// Sets the client options, overriding any already set
+    pub fn with_client_options(mut self, options: super::client::ClientOptions) -> Self {
+        self.client_options = options;
+        self
+    }
+
+    /// Sets the retry config, overriding any already set
+    pub fn with_retry_config(mut self, config: RetryConfig) -> Self {
+        self.retry_config = config;
+        self
+    }
+
+    fn get_credential_provider(&self) -> Option<CredentialProvider> {
+        if let Some(token) = self.bearer_token.as_ref() {
+            return Some(CredentialProvider::BearerToken(token.clone()));
+        }
+
+        if let (Some(client_id), Some(client_secret), Some(authority_id)) = (
+            self.client_id.as_ref(),
+            self.client_secret.as_ref(),
+            self.authority_id.as_ref(),
+        ) {
+            return Some(CredentialProvider::TokenCredential(
+                Default::default(),
+                Box::new(ClientSecretOAuthProvider::new(
+                    client_id,
+                    client_secret,
+                    authority_id,
+                    self.authority_host.as_ref(),
+                )),
+            ));
+        }
+
```

Comment:
Tried with Azure CLI auth after adding the following and it works fine.
```suggestion
+        if self.use_azure_cli {
+            return Some(CredentialProvider::TokenCredential(
+                Default::default(),
+                Box::new(AzureCliCredential::new()),
+            ));
+        }
+
```